### DISCUSSION TITLE
DataTable date filter calendar

### DISCUSSION THREAD

**Discussion by pandora-otto** [primefaces/community#3296] [PrimeReact]
I'm using a date filter on my DataTable like this:

```
<Column
    field="Modified"
    header="Modified"
    filterField="Modified"
    dataType="date"
    body={tplModifiedBody}
    sortable
    filter
    filterElement={tplModifiedFilter}></Column>
```

My tplModifiedFilter is like this:

```
const tplModifiedFilter = (options) => {
    return (
	    <Calendar
		    value={options.value}
		    onChange={(e) => options.filterCallback(e.value, options.index)}
		    dateFormat="dd/mm/yy"
		    placeholder="dd/mm/yy"
		    mask="99/99/9999"
	    />
    );
    };
```

When I select a date, for example 30/12/2024, it gives me the following:

![image](https://github.com/user-attachments/assets/be1f2b11-da70-4103-be53-58023a95642f)
![image](https://github.com/user-attachments/assets/64277681-0796-4bff-bcd9-7121e0a7c679)

If I add another rule and select another date, it gives me the following:
![image](https://github.com/user-attachments/assets/457362c4-3d39-4981-b1b9-f07be7a48412)
![image](https://github.com/user-attachments/assets/2d8e1346-a483-4871-abc5-3b6c73727a92)

So my question is, why on the second date, the format is different? How can I achieve to have the same date time format on both dates?

Any advice is much appreciated.

Thanks

**Comment by melloware** [comment]
Hmmm they both should be `Date` objects the first one should never have been a String right?  Is your JSON value `filterField="Modified"` a String or is it a Date object in the JSON?  It needs to be a Date object.

**Reply by pandora-otto** [reply]
Thank you for the message.

It is a string like below,
![image](https://github.com/user-attachments/assets/0dc4c8df-4ca8-4437-8281-b3b6acd6e075)

but I do change them to Date object and then set my data.

```
res.rows.forEach((row) => {
	row.Created = new Date(row.Created);
	row.Modified = new Date(row.Modified);
});

setData(res.rows);
```
I would expect the Calendar component to handle this.
I will try to change the json response to return Date object next and report back.

Many thanks :pray:

**Reply by melloware** [reply]
i agree let me know when you find out?

**Reply by pandora-otto** [reply]
Right! False alarm.
It turns out I was using moment library and I messed up the first date.
All is good now. Thank you and apologies for not checking thoroughly my code.
:pray:
